From: Thomas Klute <thomas2.klute@uni-dortmund.de>
Date: Sun, 22 Jan 2017 18:45:57 +0100
Subject: Test suite: Do not continue test case if Apache instance fails to
 start

On systems where namespaces aren't available, test cases in which
Apache HTTPD is expected not to start would sometimes fail when
running in parallel. The reason was a possible timing issue, where an
Apache instance for another test case might start before gnutls-cli is
run, and the TLS connection would unexpectedly succeed by connecting
to it.

Not attempting the TLS connection if HTTPD failed avoids this problem,
and also (slightly) speeds up tests.

(cherry picked from commit d39ea185bc141f880f49a68d77c1413c88fc7120)
---
 test/runtests | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/test/runtests b/test/runtests
index a253686..d530bf8 100755
--- a/test/runtests
+++ b/test/runtests
@@ -178,14 +178,16 @@ fi
 printf "TESTING: %s%s\n" "$TEST_NAME" "$EXPECTED_FAILURE"
 trap apache_down_err EXIT
 if [ -n "${USE_MSVA}" ]; then
-    MONKEYSPHERE_VALIDATION_AGENT_SOCKET="http://127.0.0.1:$MSVA_PORT" \
-					${flock_cmd} \
-					${APACHE2} -f "${t}/apache.conf" -k start \
-	|| [ -e "${t}/fail.server" ]
-else
-    ${flock_cmd} \
-	${APACHE2} -f "${t}/apache.conf" -k start \
-	|| [ -e "${t}/fail.server" ]
+    export MONKEYSPHERE_VALIDATION_AGENT_SOCKET="http://127.0.0.1:$MSVA_PORT"
+fi
+if ! ${flock_cmd} ${APACHE2} -f "${t}/apache.conf" -k start; then
+    if [ -e "${t}/fail.server" ]; then
+	echo "Apache HTTPD failed to start as expected."
+	exit 0
+    else
+	echo "Apache HTTPD unexpectedly failed to start."
+	exit 1
+    fi
 fi
 
 # check OCSP server
